pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/text/NativeTextIndexReader.java

```diff
@@ -52,7 +52,10 @@ public class NativeTextIndexReader implements TextIndexReader {
   public NativeTextIndexReader(String column, File indexDir) {
     _column = column;
     try {
-      _buffer = PinotDataBuffer.loadBigEndianFile(getTextIndexFile(indexDir));
+      String desc = "Native text index buffer: " + column;
```

Comment:
Both ways load the index off-heap. The difference is that in the original code it is loaded into direct memory, in the new code it is memory mapped. Ideally we should choose how to load the file based on the `ReadMode` (`heap` actually means direct memory), but since that cannot be easily passed to here, we may add a TODO here to address it later.

Looking at the code, I found the `_buffer` is not closed (not introduced in this PR), which can cause memory leak.

Reply:
Fixed the issue and added a todo